### PR TITLE
don't reset input value on uncontrolled input (#3122)

### PR DESCRIPTION
**Pull request by gilmeir**
### 🔦 Summary

Fixing: https://github.com/wix/wix-style-react/issues/3122

This removes a hack in the code, that causes an error when a controlled input clear button is clicked.

As part of getting rid of the hack, we aim to separate functionality between controlled / uncontrolled input:
1. No longer reset a controlled Input value when clear button is clicked. User should manage/reset value in the onClear.
2. onChange will no longer be triggered when clicking the clear button, only onClear will be called providing the clear event.

The older functionality is enabled by default for backward compatibility.
To opt in set the prop breakingChangeControlledInputOnClear={true}

### ✅ Checklist - will be updated later
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

File: src/Input/Input.spec.js [hunk 1]
```diff
@@ -583,32 +578,26 @@ describe('Input', () => {
...
         expect(await driver.hasClearButton()).toBe(true);
       });
 
       it('should not be displayed when input text is empty', async () => {
+        // controlled
         const { driver } = render(<Input value="" clearButton />);
         expect(await driver.hasClearButton()).toBe(false);
       });
 
-      it('should clear input and focus it', async () => {
```
Comment (shlomitc) [inline src/Input/Input.spec.js]: I don't think you should delete the previous behavior tests.
they should be here so we know the old behavior works.
only new tests should be applied using the new behavior

File: src/Input/Input.spec.js [hunk 2]
```diff
@@ -583,32 +578,26 @@ describe('Input', () => {
...
-      });
-
-      it('should trigger onChange on clearing as if input just emptied', async () => {
-        const onChange = jest.fn();
+      it('should trigger onClear when clicking the clear button', async () => {
+        // controlled
+        const onClear = jest.fn();
         const { driver } = render(
-          <Input onChange={onChange} value="some value" clearButton />,
+          <Input onClear={onClear} value="some value" />,
```
[the remark this answers is not in the document]
Comment (gilmeir) [reply src/Input/Input.spec.js]: Both new and old logics call onClear.
We're no longer expecting `onChange` to be called only on the new flow.
I'm adding back the old flow tests``

File: src/Input/Input.spec.js (hunk 1 above)
Comment (gilmeir) [reply src/Input/Input.spec.js]: Yep

File: src/MultiSelect/InputWithTags.spec.js [hunk 3]
```diff
@@ -19,20 +19,48 @@ describe('InputWithTags', () => {
...
 
-    expect(onChange).toHaveBeenCalledTimes(1);
-    expect(onChange.mock.calls[0][0].target.value).toBe('foo');
+      driver.inputDriver().enterText('foo');
+      expect(driver.inputDriver().getValue()).toEqual('foo');
 
-    component.clear();
+      component.clear();
+      expect(driver.inputDriver().getValue()).toEqual('');
+    });
```
Comment (gilmeir) [inline src/MultiSelect/InputWithTags.spec.js]: take back old code

File: src/MultiSelect/InputWithTags.spec.js [hunk 4]
```diff
@@ -19,20 +19,48 @@ describe('InputWithTags', () => {
...
+      );
+
+      driver.inputDriver().enterText('foo');
+      expect(onClear).toHaveBeenCalledTimes(0);
+
+      component.clear();
+      expect(onClear).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call onChange', () => {
```
Comment (gilmeir) [inline src/MultiSelect/InputWithTags.spec.js]: irrelevant

File: src/ModalSelectorLayout/ModalSelectorLayout.js [hunk 5]
```diff
@@ -293,13 +294,16 @@ export default class ModalSelectorLayout extends WixComponent {
...
-      searchValue: e.target.value,
+      searchValue,
       isSearching: true,
       items: [],
     });
-  }
+
+  _onSearchChange = e => this._updateSearchValue(e.target.value);
+
+  _onClear = () => this._updateSearchValue('');
```
[the remark this answers is not in the document]
Comment (gilmeir) [reply src/ModalSelectorLayout/ModalSelectorLayout.js]: Yes, it's equivalent to when the user triggers `onChange`: we are stating with isSearching that a new search was triggered, for rendering the loader/content

**Pull request merged**